app/models/course/group.rb

```diff
@@ -15,6 +15,57 @@ class Course::Group < ActiveRecord::Base
   validate :validate_new_users_are_unique
   validate :validate_presence_of_group_manager, on: :update
 
+  # @!attribute [r] average_experience_points
+  #   Returns the average experience points of group users in this group.
+  #   Notes:
+  #     - course_user.role == 0 is used to ensure that this applies only to students only.
```

Comment:
The thing is if your comments is far away from source code, it will be easily forget and become outdated after source code modification. and it's also not very readable since there is no context.